plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/snowflake/TestingSnowflakeServer.java

```diff
@@ -0,0 +1,95 @@
+import java.util.Optional;
+import java.util.Properties;
+import java.util.function.Function;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static io.trino.testing.TestingProperties.requiredNonEmptySystemProperty;
+
+public class TestingSnowflakeServer
+{
+    public enum TableType
+    {
+        NATIVE, ICEBERG
+    }
+
+    private static final Logger LOG = Logger.get(TestingSnowflakeServer.class);
+
+    public static final String SNOWFLAKE_JDBC_URI = requiredNonEmptySystemProperty("testing.snowflake.catalog.account-url");
+    public static final String SNOWFLAKE_USER = requiredNonEmptySystemProperty("testing.snowflake.catalog.user");
+    public static final String SNOWFLAKE_PASSWORD = requiredNonEmptySystemProperty("testing.snowflake.catalog.password");
+    public static final String SNOWFLAKE_ROLE = requiredNonEmptySystemProperty("testing.snowflake.catalog.role");
+    public static final String SNOWFLAKE_WAREHOUSE = requiredNonEmptySystemProperty("testing.snowflake.catalog.warehouse");
+    public static final String SNOWFLAKE_TEST_DATABASE = requiredNonEmptySystemProperty("testing.snowflake.catalog.database");
+
+    public void execute(String schema, String sql)
+            throws SQLException
+    {
+        checkArgument(sql != null && !sql.isEmpty(), "sql can not be null or empty");
+        executeOnDatabaseWithResultSetOperator(schema, Optional.empty(), sql);
+    }
+
+    public boolean checkIfTableExists(TableType tableType, String schema, String tableName)
+            throws SQLException
+    {
+        Function<ResultSet, Boolean> tableExistsOperator = resultSet -> {
+            try {
+                return resultSet.next();
+            }
+            catch (SQLException e) {
+                throw new RuntimeException(e);
+            }
+        };
+        String sql = "SHOW %sTABLES LIKE '%s'".formatted(tableType == TableType.ICEBERG ? "ICEBERG " : "", tableName);
+
+        return executeOnDatabaseWithResultSetOperator(schema, Optional.of(tableExistsOperator), sql).orElseThrow();
+    }
+
+    public <T> Optional<T> executeOnDatabaseWithResultSetOperator(String schema, Optional<Function<ResultSet, T>> function, String sql)
+            throws SQLException
+    {
+        Properties properties = new Properties();
+        properties.put("user", SNOWFLAKE_USER);
+        properties.put("password", SNOWFLAKE_PASSWORD);
+        properties.put("role", SNOWFLAKE_ROLE);
+        properties.put("warehouse", SNOWFLAKE_WAREHOUSE);
+        properties.put("db", SNOWFLAKE_TEST_DATABASE);
+        properties.put("schema", schema);
+
+        try (Connection connection = DriverManager.getConnection(SNOWFLAKE_JDBC_URI, properties);
+                Statement statement = connection.createStatement()) {
+            LOG.info("Using user: %s, role: %s, warehouse: %s, database: %s, schema: %s to execute: [%s]", SNOWFLAKE_USER, SNOWFLAKE_ROLE, SNOWFLAKE_WAREHOUSE, SNOWFLAKE_TEST_DATABASE, schema, sql);
```

Comment:
Is this statement needed?

Reply:
may not be important, just prints the sql along with all the vars to give context around the failure/long running queries.